### ISSUE TITLE
how to give out same kind pokemon but with different pid 

### ISSUE THREAD

**Issue by Tobyheiam** [olliz0r/Ledybot#11]
i want to give out unique one not copy of one. i make several ditto with different pid, drag these files to giveawaydeatil and set the count to 1, but this doesn't  work. only the first row, the pokemon will be given out

**Comment by olliz0r** [comment]
You are right, sadly adding the same pokemon several times to the giveaway list won't work (as it uses the dex number as key, and the key has to be unique), that's why it is ignoring the other inputs... I'll have to see what I can do about that.

**Comment by imaboy321** [comment]
If the latest update still has the folder capabilities, there should be a folder generated alongside the pk file you selected. Within that folder, you put in the different variations of that pokemon species. Based on what they name their deposit, it will select the specific variation. Ex. If someone names their deposit Adamant, it will gift them the pokemon named Adamant.pk7 within the folder, otherwise it delivers the fault.

**Comment by imaboy321** [comment]
An option to fix this is having a PID reroll option that rerolls the PID before injecting into the game.

**Comment by olliz0r** [comment]
yea I was thinking of that, too, if you want to make sure that the pokemon is not a total clone a pid reroll function would be sweet and probably not too hard to implement. Same with encryption constant.

The only issue I see is retaining shiny status, so we might have to reroll tid and then determine the pid based on that.

**Comment by k20stitch** [comment]
Guys I’ve already built a modpkm class and include the pkhex.core project as part of the solution. You can reroll pids, change stats, check legality, etc... let me know if it’s something you’re interested in

**Comment by imaboy321** [comment]
I don't think the OP means being able to edit the pk7s from within ledybot. He wants the PID to change per trade. This can easily be taken from pkhex without adding at a dependency. I'd prefer to keep minimal dependencies with a project like this as it is supposed to be standalone application.

**Comment by Mitsukiii** [comment]
```
PKHeX dump = new PKHeX();
 dump.Data = PKHeX.decryptArray(cloneshort);
dump.setRandomPID();
byte[] pkmnEdit = dump.Data;
byte[] pkmnEdited = PKHeX.encryptArray(pkmnEdit);
byte[] chkSum = BitConverter.GetBytes(PKHeX.getCHK(pkmnEdit));
Array.Copy(chkSum, 0, pkmnEdited, 6, 2);
cloneshort = pkmnEdited;
```

Should do what he want.

**Comment by Rinkoss** [comment]
@k20stitch Hi, can you still provide that modpkm class ?
